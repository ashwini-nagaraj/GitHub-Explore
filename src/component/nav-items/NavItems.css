.nav-tabs,
.nav-link.active,
.nav-tabs .nav-item.show {
  display: flex;
  align-items: center;
  justify-content: space-around;
  font-weight: bold;
  color: #000;
}
.nav-link {
  font-size: 14px;
}

.tab {
  font-size: 14px;
  color: #586069;
  text-decoration: none;
}

.search-box {
  border-radius: 5px;
  height: 39px;
  width: 300px;
  background: white;
  font-size: 15px;
  outline: none !important;
}

.input-wrapper {
  position: relative;
}
.input-icon {
  position: absolute;
  right: 10px;
  top: calc(50% - 0.5em);
}
