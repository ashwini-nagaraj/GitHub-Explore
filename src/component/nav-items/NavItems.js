import React from "react";
import "./NavItems.css";

const NavItems = ({
  user = [],
  repoSearchCallback = () => {},
  onChangeCallback = () => {},
}) => (
  <ul data-testid="nav" className="nav nav-tabs">
    <div data-testid="wrapper" class="input-wrapper">
      <input
        data-testid="search-box"
        className="search-box"
        type="text"
        placeholder="Search for any reopository and enter!"
        onChange={(e) => {
          onChangeCallback(e.target.value);
        }}
        onKeyDown={(e) => {
          if (e.keyCode === 13) {
            e.preventDefault();
            repoSearchCallback(e.target.value);
          }
        }}
      ></input>
    </div>
    <li  className="nav-item">
      <p className="nav-link mb-0 active">
        Repositories{" "}
        {user.public_repos > 1000
          ? user.public_repos.toString().charAt(0) + "K"
          : user.public_repos}
      </p>
    </li>
    <li   className="nav-item">
      <a
        target="_blank"
        rel="noopener noreferrer"
        className="tab"
        href={`https://github.com/${user.login}`}
      >
        Overview
      </a>
    </li>
    <li  data-testid="nav-item-projects" className="nav-item">
      <a
        target="_blank"
        rel="noopener noreferrer"
        className="tab"
        href={`https://github.com/${user.login}?tab=projects`}
      >
        Projects
      </a>
    </li>
  </ul>
);

export default NavItems;
