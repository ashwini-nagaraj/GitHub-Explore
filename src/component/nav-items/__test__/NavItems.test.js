import React from "react";
import ReactDOM from "react-dom";
import NavItems from "../NavItems";
import { render , cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashig", () => {
  const div = document.createElement("div");
  ReactDOM.render(<NavItems></NavItems>, div);
});

it("renders loading correctly", () => {
  const { getByTestId } = render(
    <NavItems>
    </NavItems>
  );
  const nav = getByTestId("nav");
  const wrapper = getByTestId("wrapper");
  expect(nav).toContainElement(wrapper);
});

it("matches snapshot",()=>{
    const tree=  renderer.create(<NavItems></NavItems>).toJSON();
     expect(tree).toMatchSnapshot();
 })