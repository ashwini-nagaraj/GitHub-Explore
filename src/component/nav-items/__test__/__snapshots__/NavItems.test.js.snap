// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ul
  className="nav nav-tabs"
  data-testid="nav"
>
  <div
    class="input-wrapper"
    data-testid="wrapper"
  >
    <input
      className="search-box"
      data-testid="search-box"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search for any reopository and enter!"
      type="text"
    />
  </div>
  <li
    className="nav-item"
  >
    <p
      className="nav-link mb-0 active"
    >
      Repositories
       
    </p>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="tab"
      href="https://github.com/undefined"
      rel="noopener noreferrer"
      target="_blank"
    >
      Overview
    </a>
  </li>
  <li
    className="nav-item"
    data-testid="nav-item-projects"
  >
    <a
      className="tab"
      href="https://github.com/undefined?tab=projects"
      rel="noopener noreferrer"
      target="_blank"
    >
      Projects
    </a>
  </li>
</ul>
`;
